packages/amplify-provider-awscloudformation/src/utils/hooks-manager.ts

```diff
@@ -0,0 +1,275 @@
+import { $TSAny, $TSContext, pathManager, stateManager, PathConstants, HooksConfig, skipHooks } from 'amplify-cli-core';
+import * as path from 'path';
+import _ from 'lodash';
+import ignore from 'ignore';
+import { S3 } from '../aws-utils/aws-s3';
+import * as aws from 'aws-sdk';
+import * as fs from 'fs-extra';
+import { sync } from 'glob';
+import { ProviderName } from '../constants';
+export const S3_HOOKS_DIRECTORY = 'hooks/';
+
+export async function uploadHooksDirectory(context: $TSContext): Promise<void> {
+  /**
```

Comment:
same as earlier, JS docs should be above function sig (here and everywhere else)